stdlib/public/core/StringGuts.swift

```diff
@@ -337,7 +337,8 @@ extension _StringGuts {
   where CodeUnit : FixedWidthInteger & UnsignedInteger {
     _sanityCheck(_isNative)
     _sanityCheck(CodeUnit.bitWidth == self.bitWidth)
-    return _bridgeObject(toNative: _object) as! _SwiftStringStorage<CodeUnit>
+    return Builtin.castReference(_bridgeObject(toNative: _object))
```

Comment:
CC @lorentey. BTW, this had the relatively minor effect of spewing dynamic type checks into unmanagedASCIIViews, which unfortunately gets multiplied based on that being inlined (or stopping it from profitably being so).

Reply:
D'oh! Makes perfect sense in hindsight! 👌